tensorflow/python/ops/special_math_ops.py

```diff
@@ -324,13 +324,43 @@ def sort_key(input_index, a):
     num_summed_elements = _total_size(t0_shape[-len(axes_to_sum):])
     new_shape = t0_shape[:len(preserved_axes)] + (num_broadcast_elements_t0,
                                                   num_summed_elements)
+    # Make new_shape valid, if invalid to be used in reshape
+    none_dims = [i for i, d in enumerate(new_shape)
```

Comment:
I've tried to, however I found that it is not simple task to abstract the functionalities, since the newly added components require information about different variables.
In line 327-341, it queries only the shape of `t0`, however in line 349-363, it queries both the shapes of `t0` and `t1`.
Finally, in line 373-383, it needs neither `t0` or `t1`, but the original shapes of them (before reshape operations).
If you have an idea to factor the codes out, please let me know.